### ISSUE TITLE
Tinychat enforcing HTTPS

### ISSUE THREAD

**Issue by Autotonic** [GoelBiju/pinybot#59]
any requests to http://tinychat.com need to be https now. This can cause issues on some platforms due to requests using urllib3 and how urllib3 handles https requests. Had this issues on a Debian server, not sure how it does with other platforms.

**Comment by Autotonic** [comment]
Just occurred to me that majority of users should be able to get away with changing the all the URLs beginning with `http://tinychat.com` to `https://tinychat.com` in https://github.com/GoelBiju/pinybot/blob/master/api/youtube.py

And might aswell make the referrer https in https://github.com/GoelBiju/pinybot/blob/master/api/web_request.py#L13

**Comment by zootshecfur** [comment]
yeah i had restarted my pc and had to do as u said to get it runing in again it keeped saying fialed to log in under xxx username ty and i guess everyone has to change it to https now

**Comment by Technetium1** [comment]
Thanks @Autotonic! Fix works but acinfo doesn't work even with these 3 other changes of `http://` to `https://` in tinychat_api.py: [def get_roomconfig_xml](https://github.com/GoelBiju/pinybot/blob/master/api/tinychat_api.py#L11) and [def spy_info](https://github.com/GoelBiju/pinybot/blob/master/api/tinychat_api.py#L78)

**Comment by Autotonic** [comment]
@Technetium1 try changing  https://github.com/GoelBiju/pinybot/blob/master/api/web_request.py#L13 back to http. May not have been needed in the first place

**Comment by NawtJ0sh** [comment]
yo Aida, i tested it on cloud9 ide with the changes, it still said the same thing. http://prntscr.com/dxl842

**Comment by Anonymousx12** [comment]
hey, you guys might find  this  repository usefull  since  is  the  same module https://github.com/Tinychat/Tinychat-Bot

now  if i use the pinybot  from here  https://github.com/GoelBiju/pinybot

it will just  said login failed for some reason

i did change the http to https for all tinychat apis  and links possible but still not working for me.

i think the best way to fix it is comparing both bots .

**Comment by Anonymousx12** [comment]
for me the fix was in the api directory  in  web_request.py 

line     'Referrer': 'http://tinychat.com'    changed   to      'Referrer': 'https://tinychat.com'


 line   141  changed   url = 'http://tinychat.com/login'   to  url = 'https://tinychat.com/login'
well atleast for me it works  fine 
but that should solve the issue

**Comment by Anonymousx12** [comment]
@Autotonic i think thats all we need  to do as you said is  just the https theres  no other issues

thanks a lot.

**Comment by Autotonic** [comment]
I'm still having issues running this on a few servers. If anyone has *nix VPS with python2 and wants to give running the bot a shot; please do and paste your traceback in this thread.

**Comment by Technetium1** [comment]
@Autotonic is the issue still present after https://github.com/GoelBiju/pinybot/pull/60?

**Comment by Autotonic** [comment]
For most people it should fix it. For some servers though a certificate issue is still present from requests using urllib3. Anyone running from home shouldn't have this issue though after #60 . 
Thus far i've only been able to test on my Debian VPS (which already had the same issue with YouTube Data API requests but those requests dont **force** HTTPS). Unable to test on my Arch Linux server due to Tinychat IP banning it.

Leaving this thread open for now incase anyone else runs into a any other issues relevant to this instead of making a new issue. (and for those who use pinybot but don't quite know how to navigate github and view diffs)

**Comment by weedude12** [comment]
https://gyazo.com/a482b7b1da73d160480fc4fe2651f88c 

im getting this after the https changes

**Comment by Autotonic** [comment]
@weedude12 are you running on a server?

**Comment by weedude12** [comment]
nope, from home

**Comment by GoelBiju** [comment]
Where you by any chance using the executable version of pinybot instead @weedude12 ?

**Comment by GoelBiju** [comment]
Just from reading this post on StackOverflow (http://stackoverflow.com/questions/29099404/ssl-insecureplatform-error-when-using-requests-package), you can try installing the latest version of [Python 2.7 (13)](https://www.python.org/downloads/). 

This thread isn't limited for server's, thanks for the response.

**Comment by GoelBiju** [comment]
This issue is not present in tinybot and is not (so far) an issue in the next version of pinybot. So it should be fine for now.

**Comment by weedude12** [comment]
for some reason it took a couple reinstalls for it to work again, thank you though

**Comment by Autotonic** [comment]
`pip2 install requests['security']` (escape characters as needed) seems to fix this.
